### PR TITLE
Jax gradient benchmark

### PR DESCRIPTION
**Pull request by albi3ro**
This PR adds the jax interface to the gradient benchmarking.  This means `jax` and `jaxlib` are now dependencies for the full environment.  

The benchmarks ran on my machine.

Given this data, we should really figure out why the jax interface is so incredibly slow.

```
 ========= ========== ============ ============ ============ ============
--                                        interface
-------------------- ---------------------------------------------------
 n_wires   n_layers    autograd        tf         torch         jax
========= ========== ============ ============ ============ ============
    2         3       7.19±0.3ms   23.8±0.2ms   9.98±0.2ms    217±4ms
    2         6       10.8±0.9ms   42.7±0.5ms   24.8±0.4ms    355±10ms
    5         3       15.4±0.4ms    62.7±3ms     47.0±1ms    1.04±0.01s
    5         6        25.8±1ms     120±1ms      164±4ms     1.46±0.01s
========= ========== ============ ============ ============ ============
```

### CODE REVIEW

**Comment by josh146** [comment]
@albi3ro perhaps this is a case where `@jax.jit` needs to be used to make it comparable to TF/Torch?

**Comment by albi3ro** [comment]
As expected, I tried switching it to use `@jax.jit` if the jax interface is called, and it takes even longer than before.  We are only tracking one evaluation of a derivative so we are tracking the compilation time.  

Jit gives extra execution time for the first evaluation at the benefit of subsequent executions being faster.  We aren't using subsequent executions, so we don't see the speedup.

**Comment by josh146** [comment]
That makes sense, thanks @albi3ro

**Review by josh146 — APPROVED** [review]
Looks good to me @albi3ro!
